### ISSUE TITLE
Design enhancement

### ISSUE THREAD

**Issue by ASverdlov** [ASverdlov/ray-tracing#2]
We should probably add **RenderTarget** interface class.
All possible objects to render to (like Window, Image, etc.) would inherit this interface.
And change **Render.RenderScene()** to be **Renderer.RenderScene(Scene* scene, RenderTarget* target)**.

**Comment by tulindanil** [comment]
Is it better to name it 'output'? Also we can create input interface

**Comment by ASverdlov** [comment]
I agree it is a correct generalization, but I don't see benefits of this naming. You want to move from  "Scene renders to RenderTarget" to "Input renders to Output"?
Can you elaborate more on uses of Input interface? Who would inherit it besides Scene class?
